src/main/java/seedu/address/logic/commands/AddTagTaskCommand.java

```diff
@@ -0,0 +1,149 @@
+package seedu.address.logic.commands;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_TAG;
+import static seedu.address.model.Model.PREDICATE_SHOW_ALL_TASKS;
+
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import seedu.address.commons.core.index.Index;
+import seedu.address.commons.util.ToStringBuilder;
+import seedu.address.logic.Messages;
+import seedu.address.logic.commands.exceptions.CommandException;
+import seedu.address.model.Model;
+import seedu.address.model.tag.Tag;
+import seedu.address.model.task.Note;
+import seedu.address.model.task.Status;
+import seedu.address.model.task.Task;
+import seedu.address.model.task.Title;
+
+/**
+ * Adds tags to an existing task in the address book.
+ */
+public class AddTagTaskCommand extends Command {
+
+    public static final String COMMAND_WORD = "addTagTask";
+    public static final String SHORTENED_COMMAND_WORD = "atagt";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD + " (alias: " + SHORTENED_COMMAND_WORD + ")"
+            + ": Adds new tag to the task identified "
```

Comment:
```suggestion
            + ": Adds new tags to the task identified "
```

Perhaps this should be plural as the javadoc and `MESSAGE_EDIT_TASK_SUCCESS` also refer to tags in plural?